modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormValuesJSONDeserializerTest.java

```diff
@@ -111,6 +111,61 @@ public void testDeserializationWithEmptyFields() throws Exception {
 		Assert.assertEquals("[]", value.getString(value.getDefaultLocale()));
 	}
 
+	@Test
+	public void testDeserializationWithInvalidInstanceId() throws Exception {
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm();
+
+		String serializedDDMFormValues = read(
+			"ddm-form-values-json-deserializer-invalid-instanceId.json");
+
+		DDMFormValues ddmFormValues = deserialize(
+			serializedDDMFormValues, ddmForm);
+
+		Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap =
+			ddmFormValues.getDDMFormFieldValuesMap();
+
+		Assert.assertEquals(
+			ddmFormFieldValuesMap.toString(), 4, ddmFormFieldValuesMap.size());
+
+		List<DDMFormFieldValue> text1Values = ddmFormFieldValuesMap.get(
+			"Text1");
+
+		DDMFormFieldValue text1Value = text1Values.get(0);
+
+		Assert.assertNotEquals(
+			text1Values.toString(),
+			"<script>alert(document.location)</script>",
+			text1Value.getInstanceId());
+
+		List<DDMFormFieldValue> text2Values = ddmFormFieldValuesMap.get(
+			"Text2");
+
+		DDMFormFieldValue text2Value = text2Values.get(0);
+
+		Assert.assertNotEquals(
+			text2Values.toString(),
+			"^%&214214JDJ",
+			text2Value.getInstanceId());
+
+		List<DDMFormFieldValue> select1Values = ddmFormFieldValuesMap.get(
+			"Select1");
+
+		DDMFormFieldValue select1Value = select1Values.get(0);
+
+		Assert.assertEquals(
+			select1Values.toString(), "yhar",
+			select1Value.getInstanceId());
+
+		List<DDMFormFieldValue> select2Values = ddmFormFieldValuesMap.get(
+			"Select2");
+
+		DDMFormFieldValue select2Value = select2Values.get(0);
+
+		Assert.assertEquals(
+			select2Values.toString(), "yhKiArYe",
+			select2Value.getInstanceId());
```

Comment:
@holatuwol Would it be possible to create only two variables (`List<DDMFormFieldValue> ddmFormFieldValues` and `DDMFormFieldValue ddmFormFieldValue`) and reuse them instead of creating `text1Value`, `text2Value`, `select1Value`  and `select2Value`? 

This way we keep the standard (see this other [test](https://github.com/liferay/liferay-portal/blob/337ec3a08889ac565e56de221033ea4aaa9298dc/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormValuesJSONDeserializerTest.java#L93-L106)).

Reply:
I've applied the suggested change.